crates/red_knot_python_semantic/tests/mdtest.rs

```diff
@@ -1,14 +1,22 @@
-use red_knot_test::run;
+use dir_test::{dir_test, Fixture};
 use std::path::PathBuf;
 
-/// See `crates/red_knot_test/README.md` for documentation on these tests.
-#[rstest::rstest]
-fn mdtest(#[files("resources/mdtest/**/*.md")] path: PathBuf) {
+#[dir_test(
+    dir: "$CARGO_MANIFEST_DIR/resources/mdtest",
+    glob: "**/*.md"
+)]
+fn mdtest(fixture: Fixture<&str>) {
+    // Get the file path as a string
+    let path = fixture.path();
+
+    // Convert the string path and strip the directory prefix (e.g., "resources/mdtest")
     let crate_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
         .join("resources")
         .join("mdtest")
         .canonicalize()
         .unwrap();
-    let title = path.strip_prefix(crate_dir).unwrap();
-    run(&path, title.as_os_str().to_str().unwrap());
+
+    path.strip_prefix(crate_dir.to_str().unwrap())
+        .unwrap_or(path)
+        .to_string();
 }
```

Comment:
This certainly makes the tests run a lot faster! Unfortunately, that's because you're not actually testing anything anymore 😄 You need to make sure you're actually still calling `red_knot_test::run()` on the path after you've "cleaned the path up" :-)

Reply:
Sorry, I didn't understand. Are you trying to point out it's wrong and so I need to add `red_knot_test::run()`?

Reply:
> Sorry, I didn't understand. Are you trying to point out it's wrong and so I need to add `red_knot_test::run()`?

Yes.

The way these tests work is a macro generates (at build time) a Rust test for each file in a specific directory. Each generated test runs `red_knot_test::run()` on a specific Markdown file here. The `red_knot_test::run()` call is the part of the test that actually makes assertions that red-knot is running correctly:

https://github.com/astral-sh/ruff/blob/7ca357119496ecae3bcc72549e0133e8b69c3ec6/crates/red_knot_python_semantic/tests/mdtest.rs#L13 

Your PR currently removes this call, meaning that the macro is generating a lot of tests, but those tests aren't really testing anything :-)

Reply:
Required changes done.
Thanks for review and sorry for blunders.

Reply:
No worries!